movement/io/load_bboxes.py

```diff
@@ -0,0 +1,520 @@
+    list_keys : list[str]
+        The list of keys whose values we want to extract from the literal
+        dictionaries in the `via_column_name` column.
+    cast_fn : type, optional
+        The type function to cast the values to. By default `float`.
+
+    Returns
+    -------
+    np.ndarray
+        A numpy array holding the extracted values. The rows (axis=0) are the
+        rows of the input dataframe. The columns (axis=1) follow the order of
+        the `list_attributes`.  Arrays will have at least 1 column
+        (no 0-rank numpy arrays).
+
+    """
+    list_bbox_attr = []
+    for _, row in df.iterrows():
+        row_dict_data = ast.literal_eval(row[via_column_name])
+        list_bbox_attr.append(
+            tuple(cast_fn(row_dict_data[reg]) for reg in list_keys)
+        )
+
+    bbox_attr_array = np.array(list_bbox_attr)
+
+    return bbox_attr_array
+
+
+def _ds_from_valid_data(data: ValidBboxesDataset) -> xr.Dataset:
+    """Convert a validated bboxes dataset to an xarray Dataset.
+
+    Parameters
+    ----------
+    data : movement.io.tracks_validators.ValidPosesDataset
+        The validated data object.
+
+    Returns
+    -------
+    bounding boxes dataset containing the boxes tracks,
+        boxes shapes, confidence scores and associated metadata.
+
+    """
+    # Create the time coordinate
+    time_coords = data.frame_array
+    time_unit = "frames"
+    # if fps is provided: time_coords is expressed in seconds.
+    if data.fps is not None:
+        # Compute frames from the start (first frame is frame 0).
+        # Ignoring type error because `data.frame_array` is not None after
+        # ValidBboxesDataset.__attrs_post_init__()
+        time_coords = np.arange(data.frame_array.shape[0], dtype=int)  # type: ignore
+        time_coords = time_coords / data.fps
+        time_unit = "seconds"
+
+    # Convert data to an xarray.Dataset
+    # ('time', 'individuals', 'space')
+    DIM_NAMES = tuple(a for a in MovementDataset.dim_names if a != "keypoints")
+    n_space = data.position_array.shape[-1]
+    return xr.Dataset(
+        data_vars={
+            "position": xr.DataArray(data.position_array, dims=DIM_NAMES),
+            "shape": xr.DataArray(data.shape_array, dims=DIM_NAMES),
+            "confidence": xr.DataArray(
+                data.confidence_array, dims=DIM_NAMES[:-1]
+            ),
+        },
+        # Ignoring type error because `time_coords`
+        # (which is a function of `data.frame_array`)
+        # cannot be None after
+        # ValidBboxesDataset.__attrs_post_init__()
+        coords={
+            DIM_NAMES[0]: time_coords.squeeze(),  # type: ignore
+            DIM_NAMES[1]: data.individual_names,
+            DIM_NAMES[2]: ["x", "y", "z"][:n_space],
+        },
+        attrs={
+            "fps": data.fps,
+            "time_unit": time_unit,
+            "source_software": data.source_software,
+            "source_file": None,
+            "ds_type": "bboxes",
```

Comment:
I think you should also modify the corresponding function for poses, to add a `"ds_type": "poses"` attribute.